(DOCSP-40807) Indexes > Compound indexes (#63)
mongodb/docs-cpp: source/indexes/single-field-index.txt

.. _cpp-single-field-index:

====================
Single Field Indexes
====================

.. contents:: On this page
   :local:
   :backlinks: none
   :depth: 2
   :class: singlecol

.. facet::
   :name: genre
   :values: reference
 
.. meta::
   :keywords: index, query, optimization, efficiency

Overview
--------

**Single field indexes** are indexes that hold a reference to a single field in a collection's documents. 
These indexes improve single field query and sort performance. 
They also support :manual:`TTL Indexes </core/index-ttl>`
that automatically remove documents from a collection after a certain
amount of time or at a specified clock time.

To create a single field index, call the ``create_index()`` method and specify a document containing the following information:

> - Fields on which to create the index
> - Sort order for the indexed values, ``1`` for ascending or ``-1`` for descending

Reviewer: S: same suggestion as Compound Index
```suggestion
- Fields on which to create the index.
- Sort order for the indexed values. Use ``1`` for ascending or ``-1`` for descending.
```

.. _cpp-single-field-index:

====================
Single Field Indexes
====================

.. contents:: On this page
   :local:
   :backlinks: none
   :depth: 2
   :class: singlecol

.. facet::
   :name: genre
   :values: reference
 
.. meta::
   :keywords: index, query, optimization, efficiency

Overview
--------

> **Single field indexes** are indexes that hold a reference to a single field of a
> document in a collection. These indexes improve single field query and

Reviewer: S: This wording kinda makes it seem like the index only applies to one document. You could reword to:
```suggestion
**Single field indexes** are indexes that hold a reference to a single field of a
collection's documents. These indexes improve single field query and
```

sort performance. They also support :manual:`TTL Indexes </core/index-ttl>`
that automatically remove documents from a collection after a certain
amount of time or at a specified clock time.

> To create a single-field index, call the ``create_index()`` method and specify a document containing:

Reviewer: S: stay consistent with either one word (single-field) or two (single field). Other pages are inconsistent with this, but seems like most use two words.

Also, same suggestion as above about using a full sentence to introduce a list:
```suggestion
To create a single field index, call the ``create_index()`` method and specify a document containing the following information:
```

Reviewer: applies to all mentions of "single-field"

> - The field on which to create the index
> - The sort order for the indexed values, ``1`` for ascending or ``-1`` for descending

Reviewer: S: same suggestion about articles
```suggestion
- Field on which to create the index
- Sort order for the indexed values, ``1`` for ascending or ``-1`` for descending
```

.. note::

   The default ``_id_`` index is an example of a single-field index.
   This index is automatically created on the ``_id`` field when a new
   collection is created.

Sample Data
~~~~~~~~~~~

The examples in this guide use the ``movies`` collection in the ``sample_mflix``
database from the :atlas:`Atlas sample datasets </sample-data>`. To access this collection
from your C++ application, instantiate a ``mongocxx::client`` that connects to an Atlas cluster
and assign the following values to your ``db`` and ``collection`` variables:

.. literalinclude:: /includes/indexes/indexes.cpp
    :start-after: start-db-coll
    :end-before: end-db-coll
    :language: cpp
    :dedent:

To learn how to create a free MongoDB Atlas cluster and load the sample datasets, see the
:atlas:`Get Started with Atlas </getting-started>` guide.

Create Single-Field Index
-------------------------

The following example uses the ``create_index()`` method to 
create an ascending single-field index on the ``title`` field:

.. literalinclude:: /includes/indexes/indexes.cpp
   :start-after: start-index-single
   :end-before: end-index-single
   :language: cpp
   :copyable:
   :dedent:

> The following is an example of a query that is covered by the index
> created in the preceding code example:

Reviewer: S:
```suggestion
The following query is covered by the index
created in the preceding code example:
```

.. io-code-block::
   :copyable: true

   .. input:: /includes/indexes/indexes.cpp
      :start-after: start-index-single-query
      :end-before: end-index-single-query
      :language: php
      :dedent:
   
   .. output::
      :language: cli
      :visible: false
      
      { "_id" :..., "plot" : "Peter Pan enters the nursery of the Darling children...",
      "genres" : [ "Adventure", "Fantasy", "Family" ], "runtime" : 105,...}

Additional Information
----------------------

> 

Reviewer: S: same suggestion here about grouping the Server manual links
